pkg/cache/v3/status.go

```diff
@@ -83,6 +87,37 @@ type statusInfo struct {
 	mu sync.RWMutex
 }
 
+func computeSotwStableVersion(versionMap map[string]string) string {
+	// To enforce a stable hash we need to have an ordered vision of the map.
+	keys := make([]string, 0, len(versionMap))
+	for key := range versionMap {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	mapHasher := fnv.New64()
+
+	buffer := make([]byte, 0, 8)
+	itemHasher := fnv.New64()
+	for _, key := range keys {
+		buffer = buffer[:0]
+		itemHasher.Reset()
+		itemHasher.Write([]byte(key))
+		mapHasher.Write(itemHasher.Sum(buffer))
+		buffer = buffer[:0]
+		itemHasher.Reset()
+		itemHasher.Write([]byte(versionMap[key]))
+		mapHasher.Write(itemHasher.Sum(buffer))
+	}
+	buffer = buffer[:0]
+	return hex.EncodeToString(mapHasher.Sum(buffer))
```

Comment:
nit: It took me a minute to parse that you're reusing buffer to do nothing with it.

```suggestion
	return hex.EncodeToString(mapHasher.Sum([]byte{}))
```

Reply:
The buffer being reused avoid reallocating an 8 bytes buffer for every one of those calls

Reply:
It's on the stack, so it doesn't matter, and even if you were concerned it did, you could use `[0]byte{}`, which has size 0.

Reply:
OK, scratch that actually, you can't pass an array, but you can pass nil.

Reply:
I'd expect a byte slice to never be stack allocated as go has no short string or short array optimization, but I can pass nil instead if preferred